runtime/src/leader_bank_status.rs

```diff
@@ -0,0 +1,167 @@
+use {
+    crate::bank::Bank,
+    solana_sdk::slot_history::Slot,
+    std::{
+        sync::{Arc, Condvar, Mutex, RwLock, Weak},
+        time::{Duration, Instant},
+    },
+};
+
+/// Tracks leader status of the validator node and notifies when:
+///     1. A leader slot begins
+///     2. A leader slot completes
+#[derive(Debug, Default)]
+pub struct LeaderBankStatus {
+    /// Current state of the system
+    status: Mutex<Status>,
+    /// Weak reference to the current bank
+    bank: RwLock<Option<(Slot, Weak<Bank>)>>,
```

Comment:
why these are separeted? what's wrong with below?:

```rust
slot_and_bank_with_status: Mutex<(Status, Slot, Weak<Bank>)>
```

Reply:
A lapse in judgement 😦 

Separated an internal state consisting of slot, weak_bank, status: 5cd27833c37c0ef56508a66f62a7781beed9eef3